server/bundler/loader.js

```diff
@@ -66,6 +66,7 @@ function splitTemplate( path, module, chunkName ) {
 
 	result = [
 		'page( ' + path + ', function( context, next ) {',
+		'	context.sectionRouteMatched = true;',
```

Comment:
Yes, you are right. However, I thought this is an intended behavior. When thinking about how to solve this problem, I looked into some sections and found out that they handle not found routes themselves. Take `stats` for example: the last route in `my-sites/stats/index.js` is a catch-all one, which nicely redirects a user to site-selection if no other route was matched. On the other hand, if a route in section was matched, the last route handler should **not** call the `next` function, so that no following route handlers are called, therefore terminating the request, sending the response. Try navigating to `http://calypso.localhost:3000/stats/123/12/4` for example.

But we could also add the 404 request handler after each `require()` in `require.ensure`.